layer/layer_store.go

```diff
@@ -659,15 +659,19 @@ func (ls *layerStore) initMount(graphID, parent, mountLabel string, initFunc Mou
 	}
 	p, err := ls.driver.Get(initID, "")
 	if err != nil {
+		// cleanup on failure
+		ls.driver.Remove(initID)
```

Comment:
Looks like we're not handling the error returned from these. While there may not be much we can do if this produces an error, we could at best log it as a warning.